client/web/src/components/widgets/buttons/IconButton.vue

```diff
@@ -57,10 +57,10 @@ import IconLabel from "@/components/widgets/labels/IconLabel.vue";
 
 export default defineComponent({
 	props: {
+		action: { type: Function, required: true },
```

Comment:
What would a button do if it has no action? A button's sole purpose is to do something when clicked. Otherwise we'd use an IconLabel, not an IconButton, so this prevents that misuse. Maybe you are referring to something I didn't consider?

Reply:
I suppose the main reason would be prototyping. I can't think of a web framework with a button that requires an onClick for every button, so I generally prefer omitting the callback over repeating the code for an empty arrow function in all our (many) unused buttons. I would be satisfied deferring to another reviewer, though.

Reply:
One other benefit is that it directs developers to not set an `@click="..."` directly on the component, which is bad practice but a very easy mistake to make. So easy, in fact, that I just found exactly that issue in the code I wrote for this commit! `Document.vue:110-111` has the following:

```vue
<IconButton :action="() => _" @click="swapWorkingColors" :icon="'Swap'" title="Swap (Shift+X)" :size="16" />
<IconButton :action="() => _" @click="resetWorkingColors" :icon="'ResetColors'" title="Reset (Ctrl+Shift+X)" :size="16" />
```

That happened as a result of porting over functionality to use `:action` in bulk, and not looking closely enough. But in the future someone who adds `@click="..."` and then notices Vue complaining with a warning in the console will realize they need to switch to the `:action="..."` syntax which properly uses the Vue click binding on the `<button>` element rather than the entire component.

I think adding `:action="() => _"` to squelch the warning during prototyping is better practice, and it serves as a reminder in the code that it needs to be implemented instead of easily forgetting about a button that has no functionality.